xarray/core/computation.py

```diff
@@ -1338,3 +1339,69 @@ def polyval(coord, coeffs, degree_dim="degree"):
         coords={coord.name: coord, degree_dim: np.arange(deg_coord.max() + 1)[::-1]},
     )
     return (lhs * coeffs).sum(degree_dim)
+
+
+def _calc_idxminmax(
+    *,
+    array,
+    func: Callable,
+    dim: Optional[Hashable],
+    skipna: bool = None,
+    fill_value: Any = None,
+    keep_attrs: bool = None,
+    **kwargs: Any,
+):
+    """Apply common operations for idxmin and idxmax."""
+    # This function doesn't make sense for scalars so don't try
+    if not array.ndim:
+        raise ValueError("This function does not apply for scalars")
+
+    if dim is not None:
+        pass  # Use the dim if available
+    elif array.ndim == 1:
+        # it is okay to guess the dim if there is only 1
+        dim = array.dims[0]
+    else:
+        # The dim is not specified and ambiguous.  Don't guess.
+        raise ValueError("Must supply 'dim' argument for multidimensional arrays")
+
+    if dim not in array.dims:
+        raise KeyError(f'Dimension "{dim}" not in dimension')
+    if dim not in array.coords:
+        raise KeyError(f'Dimension "{dim}" does not have coordinates')
+
+    # These are dtypes with NaN values argmin and argmax can handle
+    na_dtypes = "cfO"
```

Comment:
`datetime64` and `timedelta64` also have missing values. Those have type codes `m` and `M`.

To encourage consistency, would you mind moving this into a constant in `dtypes.py`? e.g.,
```python
NA_DTYPE_KINDS = 'cfOmM'
```

Reply:
Those aren't supported by `argmin` and `argmax` yet so I can't support them here, either.

Reply:
Are timedelta/datetime dtypes just broken?

Reply:
They work without NaT values, I specifically test them.  But NaT values don't work.